core/cppgc.rs

```diff
@@ -391,12 +391,71 @@ pub fn try_unwrap_cppgc_object<'sc, T: GarbageCollected + 'static>(
     return None;
   }
 
-  Some(Ptr {
+  Some(UnsafePtr {
     inner: obj,
     root: None,
   })
 }
 
+pub struct Ref<T: GarbageCollected> {
+  inner: v8::cppgc::Persistent<CppGcObject<T>>,
+}
+
+impl<T: GarbageCollected> std::ops::Deref for Ref<T> {
+  type Target = T;
+  fn deref(&self) -> &T {
+    &self.inner.borrow().unwrap().member
+  }
+}
+
+#[doc(hidden)]
+#[allow(clippy::needless_lifetimes)]
+pub fn try_unwrap_cppgc_persistent_object<
```

Comment:
this is a bit duplicative. using the existing unwrap api to UnsafePtr and then creating a Persistent from that is fine.